elide-spring/elide-spring-boot-autoconfigure/src/test/java/example/tests/ControllerTest.java

```diff
@@ -221,26 +231,499 @@ public void jsonForbiddenApiPatchTest() {
+                                                type("group"),
+                                                id("com.example.operationsinfer2")
+                                             )
+                                )
+                        )
+                )
+                .when()
+                .post("/json/operations")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .extract();
+        String result = deleteResponse.asString();
+        String expected = """
+                {"atomic:results":[{"data":null},{"data":null}]}""";
+        assertEquals(expected, result);
+    }
+
+    @Test
+    public void jsonApiAtomicOperationsExtensionRelationshipTest() {
+        ExtractableResponse<Response> response = given()
+                .contentType(JsonApiController.JSON_API_ATOMIC_OPERATIONS_CONTENT_TYPE)
+                .accept(JsonApiController.JSON_API_ATOMIC_OPERATIONS_CONTENT_TYPE)
+                .body(
+                        atomicOperations(
+                                atomicOperation(AtomicOperationCode.add,
+                                        datum(resource(
+                                                type("group"),
+                                                id("com.example.operationsrel1"),
+                                                attributes(
+                                                        attr("commonName", "Foo1")
+                                                )
+                                        ))
+                                ),
+                                atomicOperation(AtomicOperationCode.add,
+                                        datum(resource(
+                                                type("maintainer"),
+                                                id("com.example.person1"),
+                                                attributes(
+                                                        attr("commonName", "Person1")
+                                                )
+                                        ))
+                                ),
+                                atomicOperation(AtomicOperationCode.add, "/group/com.example.operationsrel1/products",
+                                        datum(resource(
+                                                type("product"),
+                                                id("com.example.operations.product1"),
+                                                attributes(
+                                                        attr("commonName", "Product1")
+                                                )
+                                        ))
+                                ),
+                                atomicOperation(AtomicOperationCode.add,
+                                        ref(
+                                                type("group/com.example.operationsrel1/products"),
```

Comment:
Is this a valid way of referencing type?  This looks more like a href.

Reply:
I'm actually not sure. I thought the intent was that the ref could always be expressed as a href and vice-versa. It's not really that clear from the spec why there were 2 ways to express the target of the operation.

Reply:
My take is that some/most JSON-API APIs only expose models at the root of the URL path (/books, /invoices).  Then the REF objects can be used to navigate to most things.  

The type of the REF then is really the root collection.

Reply:
I have updated the test case to use the href instead of the ref.